### PR TITLE
Version 5.1

### LINKED ISSUE #1559: K version bump feature list (5.1.0)
- [x] `claim` keyword
- [x] `macro` expansion with Bison parser
- [x] Switch to `kx` pipeline by default (KEVM, KIELE using kx)
- [x] Semantic versioning setup: #1857

### PR DESCRIPTION
**Pull request by ehildenb**
Fixes: #1559

- [x] File `./package/version` has been bumped to `NEW_MAJOR.NEW_MINOR.0`, and `./package/version.sh sub` has been invoked.
- [x] `CHANGELOG.md` has been updated with new features.
- [x] `krun` help script is correctly reflecting the available options
- [x] All `*Options.java` files are up-to-date.
- [x] K active projects page is up-to-date (receiving commits in the last 3 months).
- [x] Audit of warnings in regression test-suite and tutorial.
- [x] Audit package dependencies in `package/`.

### CODE REVIEW

**Comment by ehildenb** [comment]
To "Audit warnings/errors", I just grepped for `[Warning]` and `[Error]` in the test-suites and checked that each occurence made sense for the test.

To "Audit package depnedencies", I:

-   organized/sorted the lists of package dependencies in each package,
-   changed from using `lib*-dev` packages on debian-based distros to the proper `lib*-NNN` packages, and
-   quickly scanned through for anything obvious that could be removed (didn't find anything).

**Comment by ehildenb** [comment]
Evidence this works: https://office.runtimeverification.com/jenkins/blue/organizations/jenkins/k/detail/PR-1921/11/pipeline/

File: k-distribution/src/main/scripts/bin/krun [hunk 1]
```diff
@@ -119,8 +118,6 @@ present; see below).
                            stdin/stdout via cells with "stream" attribute.
                            Defaults to enabled when performing concrete
                            execution, and disabled when --search is passed.
-      --no-substitution-filtering  Don't filter conjuncts with anonymous
```
Comment (dwightguth) [inline k-distribution/src/main/scripts/bin/krun:122]: Why was this removed?

**Review by dwightguth — COMMENTED** [review]
Did you actually change any of the package dependencies, or just reorder them?

File: k-distribution/src/main/scripts/bin/krun (hunk 1 above)
Comment (ehildenb) [reply k-distribution/src/main/scripts/bin/krun:122]: It's not used anywhere as far as I can tell. It's not even passed into any sub-commands as far as I can tell.

**Comment by ehildenb** [comment]
Nope, didn't end up removing any dependencies, just re-ordered and tried to change some `lib*-dev` to `lib*NNN`, but turns it it didn't work for those ones.

File: k-distribution/src/main/scripts/bin/krun (hunk 1 above)
Comment (dwightguth) [reply k-distribution/src/main/scripts/bin/krun:122]: It was supposed to get passed to kprint. I agree it's not currently. That's a bug we should make an issue for, but I don't think it makes sense to delete it in this pr.

**Review by dwightguth — APPROVED** [review]
I am approving, but please bear in mind what I said about the substitution filtering flag; it may be more appropriate to leave it there since we are going to be making it function in a future pr. I will leave it to your judgement though.
